src/app/clusters/groups-server/groups-server.cpp

```diff
@@ -218,7 +219,7 @@ bool emberAfGroupsClusterViewGroupCallback(chip::app::Command * commandObj, Grou
         VerifyOrExit((writer = commandObj->GetCommandDataElementTLVWriter()) != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
         SuccessOrExit(err = writer->Put(TLV::ContextTag(0), status));
         SuccessOrExit(err = writer->Put(TLV::ContextTag(1), groupId));
-        SuccessOrExit(err = writer->PutString(TLV::ContextTag(2), reinterpret_cast<char *>(&groupName[1]), groupName[0]));
+        SuccessOrExit(err = writer->Put(TLV::ContextTag(2), groupName));
```

Comment:
This is changing the encoding from string to octet string, in terms of the actual type in the TLV and hence what the other side will interpret this as.  Is this intended?  

If we're going to represent strings as `ByteSpan` (which I am stll not convinced about; `Span<char>` or the like would make way more sense to me), we should have a `PutString` overload that takes `ByteSpan`.